### ISSUE TITLE
3.2.2 Beta IK is deformed

### ISSUE THREAD

**Issue by NHodgesVFX** [godotengine/godot#38183]
**Godot version:**
3.2.2 Beta

**OS/device including version:**
Windows 10 64bit

**Issue description:**

![IKIssue](https://user-images.githubusercontent.com/11403345/80235471-c1b60700-8627-11ea-89e7-d6ab92a717db.PNG)

IK is not deforming correctly. If you look at either the mesh or the wire frame of the bones they are deformed incorrectly 

This is similar to https://github.com/godotengine/godot/issues/37095 I tried the file in this post https://github.com/godotengine/godot/issues/37095#issuecomment-599859123 but it seemed fine.

**Steps to reproduce:**

1. Import an object with a bone chain suitable for IK
2. In Godot add ik node, set it up with a target.
3. press play ik, bones are now incorrectly deformed

**Minimal reproduction project:**
Press "PlayIK"

[IK.zip](https://github.com/godotengine/godot/files/4530176/IK.zip)

**Comment by aaronfranke** [comment]
Does it work in 3.2.1 or another release?

**Comment by NHodgesVFX** [comment]
> Does it work in 3.2.1 or another release?

I recreated the project in godot 3.1.2, and it works

**Comment by Zireael07** [comment]
I think it was already reported as not working in 3.2.1, though?

**Comment by aaronfranke** [comment]
@Zireael07 If you can find a duplicate, that would be great, but I can't find any.

**Comment by NHodgesVFX** [comment]
> I think it was already reported as not working in 3.2.1, though?

It was, but there was a pull to fix it and it was closed. However, I think this is a diffrent problem with the same symptoms. As I tried a test file from the other issue and it seemed to work.

**Comment by WilliamTambellini** [comment]
@NHodgesVFX  could you please just advice if that bug has infected the official 3.2.2 ? Kind

**Comment by NHodgesVFX** [comment]
@WilliamTambellini yes this still happens in the official 3.2.2, it also happens in the 3.2.3rc4

**Comment by akien-mga** [comment]
CC @TwistedTwigleg @AndreaCatania @JFonS. Does #40774 help with this issue?

**Comment by TwistedTwigleg** [comment]
@akien-mga - It doesn't appear that #40774 helps with this issue, at least not with how it is currently written.

I'm currently working on trying to find a better fix for the root bone not rotating issue, and will try the project in this issue again when I've found a solution.

**Comment by WilliamTambellini** [comment]
Update : 3.2.3 seems to also suffer from that bug.

**Comment by josephbburg** [comment]
Is this the same bug as https://github.com/godotengine/godot/issues/40098 ?
I have noticed this bug in the 3.x branch, it only seems to affect the first bone in the chain.

**Comment by donn-xx** [comment]
FWIW, I am seeing the same thing in Godot 4 Beta 3 (October 2022). Weird deforms of the mesh - as well as bones that are kind of bent on themselves. It's hard to describe.

Have a look at the yellow RightUpperLeg bone in this screen shot.
![image](https://user-images.githubusercontent.com/392218/196257920-05f29a6d-e32d-4ea6-ba4b-42599f3430d7.png)

**Comment by Zireael07** [comment]
@donn-xx What format was this imported from?

**Comment by donn-xx** [comment]
.glb from Blender 3.3 (not .gltf but not for any reason other than size)

BTW - The 3DCCIK modification in Godot 4 beta does not deform the bones.

**Comment by donn-xx** [comment]
I made a small demo project (Godot Beta 3) and included the Blend source file. Open the project, choose the IK Node and press its play button to see the weird:
[fabrikbug_4beta3_demo.zip](https://github.com/godotengine/godot/files/9806621/fabrikbug_4beta3_demo.zip)

**Comment by Zireael07** [comment]
@donn-xx I believe there's an open issue for bones deforming in Godot 4 beta for gltf, and I guess glb also counts

**Comment by donn-xx** [comment]
Just a note - the skeleton etc. seem fine after import. It's only when the Fabrik IK is used that they go funny. Same skelington, same project, using 3dcccik also works without deforming.

**Comment by donn-xx** [comment]
Here's another screenshot. I tried the IK node and I got spaghetti. Player is via a .glb file from Blender.
![image](https://user-images.githubusercontent.com/392218/198822995-1f2eb094-230c-4d5b-932b-89ba7526a730.png)

**Comment by donn-xx** [comment]
New finding:
If you *do not* make a bone map during import, then the IK starts working. 

I made a new model with Blender and auto made a metarig (human) and then took that out to Godot 4b3. I wanted to see whether the bone rotations as set automatically would work or not.
The IK failed anyway. Then I thought about it and decided to remove the bone mapping. I had to rewire the IK, but it works!

So, that's a data point for debugging, I hope.
